src/main/java/org/swisspush/reststorage/FileSystemStorage.java

```diff
@@ -64,11 +66,16 @@ public void get(String path, String etag, final int offset, final int count, fin
                     if (props.isDirectory()) {
                         fileSystemDirLister.handleListingRequest(path, offset, count, handler);
                     } else if (props.isRegularFile()) {
-                        fileSystem().open(fullPath, new OpenOptions(), event1 -> {
+                        fileSystem().open(fullPath, OPEN_OPTIONS_READ_ONLY, event1 -> {
                             DocumentResource d = new DocumentResource();
-                            d.length = props.size();
-                            d.readStream = event1.result();
-                            d.closeHandler = v -> event1.result().close();
+                            if (event1.failed()) {
+                                d.error = true;
+                                d.errorMessage = event1.cause().getMessage();
```

Comment:
Should we also `logger.debug( "Failed to open file for read" , event1.cause() );` to also have stack trace when debugging related problems?